pkg/webhook/mutation/pod_mutator/pod_mutator.go

```diff
@@ -150,8 +151,19 @@ func (webhook *podMutatorWebhook) handlePodMutation(ctx context.Context, mutatio
 	ctx, span := dtotel.StartSpan(ctx, webhookotel.Tracer(), spanOptions()...)
 	defer span.End()
 
+	needsInjection := false
```

Comment:
I would extract this logic and use a new function instead of making handePodMutation even bigger